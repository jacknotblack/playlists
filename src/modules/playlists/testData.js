export const playLists = [
  {
    id: 0,
    name: "aaa",
    songs: [0, 3]
  },
  {
    id: 1,
    name: "bbb",
    songs: []
  },
  {
    id: 2,
    name: "ccc",
    songs: [2]
  },
  {
    id: 3,
    name: "ddd",
    songs: [3]
  },
  {
    id: 4,
    name: "zzz",
    songs: []
  },
  {
    id: 5,
    name: "zzzz",
    songs: []
  },
  {
    id: 6,
    name: "abc",
    songs: []
  },
  {
    id: 7,
    name: "abczzz",
    songs: []
  },
  {
    id: 8,
    name: "123",
    songs: []
  }
];

export const songs = [
  { id: 0, title: "abc1", artist: "Jack", duration: "1:28" },
  { id: 1, title: "abc2", artist: "Tyler", duration: "1:23" },
  { id: 2, title: "abc3", artist: "Tyler", duration: "2:33" },
  { id: 3, title: "abc4", artist: "Randy", duration: "5:41" },
  { id: 4, title: "abc5", artist: "Tyler", duration: "1:34" },
  { id: 5, title: "abc6", artist: "Jack", duration: "4:13" },
  { id: 6, title: "abc7", artist: "Jack", duration: "3:25" },
  { id: 7, title: "abc8", artist: "Randy", duration: "2:53" },
  { id: 8, title: "abc9", artist: "Jack", duration: "2:27" },
  { id: 9, title: "abc10", artist: "Randy", duration: "4:23" },
  { id: 10, title: "abc11", artist: "Tyler", duration: "3:33" }
];
