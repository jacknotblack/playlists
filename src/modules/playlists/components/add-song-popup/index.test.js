import React from "react";
import { shallow } from "enzyme";
import AddSongPopup from "./index";
import { playLists } from "../../testData";

const setup = props => {
  const enzymeWrapper = shallow(<AddSongPopup {...props} />);
  return {
    props,
    enzymeWrapper
  };
};

describe("<AddSongPopup />", () => {
  const initProps = {
    playLists,
    songID: 3,
    addSongToList: jest.fn(),
    onClosing: jest.fn()
  };
  describe("render", () => {
    const { enzymeWrapper } = setup(initProps);

    it("renders without crashing", () => {
      expect(enzymeWrapper.length).toBe(1);
    });
    describe("render playList", () => {
      describe("playList length = 0", () => {
        it("should match snapshot", () => {
          const { enzymeWrapper } = setup({
            ...initProps,
            playLists: []
          });
          expect(enzymeWrapper).toMatchSnapshot();
        });
      });

      describe("playList is undefined", () => {
        const { enzymeWrapper } = setup({ ...initProps, playLists: undefined });
        it("should match snapshot", () => {
          expect(enzymeWrapper).toMatchSnapshot();
        });
      });
    });
  });

  describe("function props", () => {
    const { enzymeWrapper, props } = setup(initProps);

    it('should trigger "onClosing" when closing button is clicked', () => {
      enzymeWrapper
        .find(".close-button")
        .get(0)
        .props.onClick();
      expect(props.onClosing).toHaveBeenCalled();
    });

    it('should trigger "onClosing"/"addSongToList" when a song is clicked', () => {
      enzymeWrapper
        .find(".list-item")
        .get(2)
        .props.onClick();
      expect(props.onClosing).toHaveBeenCalled();
      expect(props.addSongToList).toHaveBeenCalledWith(2, 3);
    });
    it('should trigger "onClosing" and not trigger"addSongToList" when a song is clicked and it already on the list', () => {
      enzymeWrapper
        .find(".list-item")
        .get(3)
        .props.onClick();
      expect(props.onClosing).toHaveBeenCalled();
      expect(props.addSongToList).not.toHaveBeenCalled();
      expect(window.alert).toHaveBeenCalledWith(
        "this song is on the list already"
      );
    });
  });
});
