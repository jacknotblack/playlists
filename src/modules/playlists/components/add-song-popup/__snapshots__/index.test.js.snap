// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddSongPopup /> render render playList playList is undefined should match snapshot 1`] = `
<div
  className="add-song-popup"
>
  <div
    className="playlists-window"
  >
    <span
      className="close-button"
      onClick={[MockFunction]}
    >
      X
    </span>
    <br />
    <span>
      There are no playlist yet
    </span>
  </div>
</div>
`;

exports[`<AddSongPopup /> render render playList playList length = 0 should match snapshot 1`] = `
<div
  className="add-song-popup"
>
  <div
    className="playlists-window"
  >
    <span
      className="close-button"
      onClick={[MockFunction]}
    >
      X
    </span>
    <br />
    <span>
      There are no playlist yet
    </span>
  </div>
</div>
`;
